Utilities/WebAppCleanUp/WebAppCleanUp.groovy

```diff
@@ -0,0 +1,124 @@
+// run setup tasks
+setup(args)
+
+// process build group list
+println "** Deleting build groups: $groups"
+groups.each { group ->
+  String name = group.trim()
+  repoClient.deleteBuildResults(name)
+  println "* Deleting build group '$name' -> Status = ${repoClient.getLastStatus()}"
+}
+
+// process collections list
+println "** Deleting collections: $collections"
+collections.each { collection ->
+  String name = collection.trim()
+  repoClient.deleteCollection(name)
+  println "* Deleting collection '$name' -> Status = ${repoClient.getLastStatus()}"
+}
+
+// end script
+
+/*
+  setup :
+  handle cli arguments, load WebAppCleanUp.properties file, 
+  create repository client, populate deletion lists
+*/
+def setup(String[] args) {
+
+    // parse input arguments
+	String usage = 'WebAppCleanUp.groovy [options]'
+	String header = 'options:'
+	
+	def cli = new CliBuilder(usage:usage,header:header)
+	cli.g(longOpt:'groups', args:1, 'Comma separated list of Build Groups to delete')
+	cli.c(longOpt:'collections', args:1, 'Comma separated list of Collections to delete')
+	cli.G(longOpt:'groupsFile', args:1, 'Absolute or relative path (from script directory) of file containing build groups to delete')
+	cli.C(longOpt:'collectionsFile', args:1, 'Absolute or relative path (from script directory) of file containing collections to delete')
+	
+	// web application credentials (overrides properties in WebAppCleanUp.properties)
+	cli.u(longOpt:'url', args:1, 'DBB WebApp URL')
+	cli.i(longOpt:'id', args:1, 'DBB WebApp ID')
+	cli.p(longOpt:'pw', args:1,  'DBB WebApp Password')
+	cli.P(longOpt:'pwFile', args:1, 'Absolute or relative (from this script) path to file containing DBB password')
+	cli.h(longOpt:'help', 'Prints this message')
+	
+	def opts = cli.parse(args)
+	if (!args || !opts) {
+	    cli.usage()
+		System.exit(1)
+	}
+	
+	// if help option used, print usage and exit
+    if (opts.h) {
+		cli.usage()
+		System.exit(0)
+	}
+
+    // load user properties file
+   	properties.load(new File("${getScriptDir()}/user.properties"))
+   	
+   	// update authentication properties with cli options
+   	if (opts.u) properties.url = opts.url
+   	if (opts.i) properties.id = opts.id
+   	if (opts.p) properties.pw = opts.pw
+   	if (opts.P) properties.pwFile = opts.pwFile
+   	
+   	// validate authentication properties
+   	assert properties.url : "Missing 'url' argument/property"
+   	assert properties.id : "Missing 'id' argument/property"
+   	if (!(properties.pw || properties.pwFile)) {
+   		assert properties.pw : "Missing 'pw' argument/property"
+   		assert properties.pwFile : "Missing 'pwFile' argument/property"
+   	}
+   	
+   	//create repository client
+   	println "Creating repository client for ${properties.url}"
+   	repoClient = new RepositoryClient().forceSSLTrusted(true).url(properties.url).userId(properties.id)
+   	if (properties.pw)
+   	   repoClient.setPassword(properties.pw)
+   	else
+   	   repoClient.setPasswordFile(properties.pwFile)
```

Comment:
I don't think that this API exists. The RepositoryClient accepts a java.io.File, but not a String value
```
public void setPasswordFile(java.io.File passwordFile)
```